src/Hedgehog.Experimental/Gen.fs

```diff
@@ -423,17 +423,46 @@ module GenX =
                 else
                   Gen.constant (None: 'a option) |> wrap}
 
-        | Shape.Array s when s.Rank = 1 ->
+        | Shape.Array s ->
             s.Element.Accept {
               new ITypeVisitor<Gen<'a>> with
               member __.Visit<'a> () =
+                let newMultidimentionalArray (lenghts: int list) =
+                  let array = lenghts |> Array.ofList
+                  System.Array.CreateInstance (typeof<'a>, array)
+                let setMultidimentionalArrayEntries (data: 'a seq) maxIndices (array: Array) =
+                  let currentIndices = Array.create (List.length maxIndices) 0
+                  use en = data.GetEnumerator ()
+                  let rec loop currentIndicesIndex = function
+                    | [] ->
+                        en.MoveNext () |> ignore
+                        array.SetValue(en.Current, currentIndices)
+                    | currentMaxIndex :: remaningMaxIndices ->
+                        for i in 0..currentMaxIndex - 1 do
+                          currentIndices.[currentIndicesIndex] <- i
+                          loop (currentIndicesIndex + 1) remaningMaxIndices
+                  loop 0 maxIndices
```

Comment:
I feeeeeel like `setMultidimentionalArrayEntries` is complex enough to deserve its own bespoke tests, but whatever lol.

Reply:
I simplified it a bit by making it pure (instead of overwriting all the entries of a given array, it now creates that array and returns it).  I also tried to make it simpler by improving names.  In particular, I am now using the name `lengths` everywhere for the same thing.  I picked that name because it is the name used in the .NET library function `Array.CreateInstance (Type, int[])`.

I added a test for `MultidimensionalArray.createWithGivenEntries`.

What do you think?

Reply:
Resolved.